src/SFML/Audio/SoundRecorder.cpp

```diff
@@ -202,6 +213,32 @@ const std::string& SoundRecorder::getDevice() const
 
 
 ////////////////////////////////////////////////////////////
+void SoundRecorder::setChannelCount(unsigned int channelCount)
+{
+    if (m_isCapturing)
+    {
+        sf::err() << "It's not possible to change the channels while recording. Please stop the recording and call setChannelCount() again." << std::endl;
+        return;
+    }
+
+    if (channelCount > 2)
```

Comment:
I'm generally not sure if it is helpful to provide the user with a different channel count than specified... Maybe assert could be considered, to make the logic error (invalid argument) more noticeable.